querybook/webapp/components/Search/SearchResultItem.tsx

```diff
@@ -47,6 +52,93 @@ function openClick(
     }
 }
 
+interface IQueryItemProps {
+    preview: IQueryPreview;
+    searchString: string;
+    environmentName: string;
+}
+
+export const QueryItem: React.FunctionComponent<IQueryItemProps> = ({
+    preview,
+    environmentName,
+    searchString,
+}) => {
+    const {
+        author_uid: authorUid,
+        created_at: createdAt,
+        engine_id: engineId,
+        id,
+        query_text: queryText,
+        title,
+    } = preview;
+    const { userInfo: authorInfo, loading } = useUser({ uid: authorUid });
+
+    const isQueryCell = preview.query_type === 'query_cell';
+
+    const url = isQueryCell
+        ? `/${environmentName}/datadoc/${preview.data_doc_id}/?cellId=${id}`
+        : `/${environmentName}/query_execution/${id}/`;
+    const handleClick = React.useMemo(() => openClick.bind(null, url), [url]);
+    const queryEngineById = useSelector(queryEngineByIdEnvSelector);
+
+    if (loading) {
+        return <div className="SearchResultItem QueryItem">Loading...</div>;
+    }
+
+    // Query cell title is data cell title
+    // Query execution title is "<title> > Execution <id>" for data cell executions
+    // or "Adhoc Execution <id>" for adhoc exeuctions
+    const resultTitle = isQueryCell
+        ? title ?? 'Untitled'
+        : `${title != null ? `${title} >` : 'Adhoc'} Execution ${id}`;
+    const queryTextHighlightedContent = (preview.highlight || {}).query_text;
+    const getQueryTextHighlightedDOM = (queryTextContent: string[]) => (
+        <span
+            className="result-item-description"
+            dangerouslySetInnerHTML={{
+                __html: formatHighlightStrings(queryTextContent),
+            }}
+        />
+    );
+
+    const queryEngine = queryEngineById[engineId];
+    const language = queryEngine?.language ?? 'presto';
+    const formattedQuery = format(queryText, language, {
```

Comment:
formatting per render is kinda expensive,  i dont think its needed for search. If you want to keep it can you make ThemedCodeHighlight support auto formatting?

Reply:
removed it